code/game/turfs/unsimulated/floor.dm

```diff
@@ -77,6 +77,29 @@
 			if((direction & junction) == direction)
 				overlays += dirtlayers["diag[direction]"]
 				dircount += 1
+	if(!(dirtlayers.len))
+		for(var/dirtdir in alldirs)
+			dirtlayers["side[dirtdir]"] = image('icons/turf/newsnow.dmi',"snowpath-Side",dir = dirtdir)
+		for(var/diagdir in diagonal)
+			dirtlayers["diag[diagdir]"] = image('icons/turf/newsnow.dmi',"dirtquarter",dir = diagdir,layer=2.1)
+			dirtlayers["snow[diagdir]"] = image('icons/turf/newsnow.dmi',"snowpath",dir = diagdir)
+		for(var/dirtdir in cardinal)
+			dirtlayers["snow[dirtdir]"] = image('icons/turf/newsnow.dmi',"snowpath-half",dir = dirtdir)
+			var/realdir = null
+			switch(dirtdir)
+				if(NORTH)
+					realdir = EAST|SOUTH|WEST
+				if(SOUTH)
+					realdir = WEST|NORTH|EAST
+				if(EAST)
+					realdir = SOUTH|WEST|NORTH
+				if(WEST)
+					realdir = NORTH|EAST|SOUTH
+			dirtlayers["snow[realdir]"] = image('icons/turf/newsnow.dmi',"snowpath-TJunction",dir = dirtdir)
+		dirtlayers["snow15"] = image('icons/turf/newsnow.dmi',"snowpath-Crossroads")
+		dirtlayers["snow0"] = image('icons/turf/newsnow.dmi',"snowpath-circle")
+		dirtlayers["snow3"] = image('icons/turf/newsnow.dmi',"snowpath",dir = 1)
+		dirtlayers["snow12"] = image('icons/turf/newsnow.dmi',"snowpath",dir = 8)
 	if(dircount == 4)
```

Comment:
VERY rough copypaste of the pathcode, moving it from the base snow turf to undersnow

This managed to cut a grand total of 10 SECONDS off the load time. Down from 4 min 10 to 4

It's a start